app/controllers/users_controller.rb

```diff
@@ -67,28 +67,48 @@ def list_pending_requested
     @roles = Role.all
   end
 
+
+  # Differences between show _selection and show:
+  #   The key difference between show_selection and the show methods is in how they determine whether the current user has the authority to delete/edit a selected user
+  #   The show method only checks if the current user is not a student or him/herself. It assumes that all other roles are authorised to edit/delete the information of all users.
+  #   The show_selection method allows access only if the current user is higher up the hierarchy of roles than the user s/he is requesting to edit.
+  # The show method additionally calculates the number of total users and the number of groups the user is participating in.
+  # These numbers are used in the show view to display appropriate warning messages when deleting a user.
+
+  # determines if the current user is authorised to see/edit the information of the user in params.
+  # the test used to determine is whether the current user is higher up the hierarchy of roles than the user s/he is requesting to edit.
+  # If these permissions check out, the user is redirected to the 'show' view
+  # If these checks come out negative, the user to given an error message and redirected to a list of all users.
   def show_selection
     @user = User.find_by(name: params[:user][:name])
+    #make sure that the requested user exists
```

Comment:
I don't think one line comments like this (and on line 91, 95, 109 and 285) would be necessary. The code beneath this comment is very straightforward. 

However, comments like the ones on line 87 and 288 are useful as it is not immediately clear what the lines of code beneath them do.

Reply:
This has been fixed for 91 and 95

Reply:
All have been fixed.